### DISCUSSION TITLE
React/Vite project gives 404 not found on Vercel

### DISCUSSION THREAD

**Discussion by ol93** [community/community#116416] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

I'm a developer who is just starting to get into web development. I'm trying to deploy my first project on Vercel but it gives me the 404 not found error and I don't know what I am doing wrong. When I host it locally everything works fine. This is my repo: https://github.com/ol93/handiDraw.

**Comment by Vanblerk666** [comment]
i've got the same issue - did you get it resolved?

**Reply by ol93** [reply]
Really frustrating isn’t it?! I got it resolved, what worked for me is to make sure that the file structure is the way it’s supposed to be. Apparently I moved the public folder where it wasn’t supposed to be and that was causing all types of problems. I’ve read elsewhere that creating a vercel doc that reroutes every route to / may also help. For me it was the first one tho. Hope this helps, let me know!

**Comment by kanetjuh** [comment]
What page gives a `not found` error?

**Comment by gurjeetsinghvirdee** [comment]
hey @ol93 

rewrite your `vercel.json` file like this, maybe it help!

```js
{
  "rewrites": [{ "source": "/(.*)", "destination": "/index.html" }]
}
```

or you can check your **vercel deploy** logs it will show you why it will give 404

**Reply by piyushyadav0191** [reply]
thanks, It works
